docs/rfcs/2025-03-17-compute-prewarm.md

```diff
@@ -0,0 +1,345 @@
+# Compute rolling restart with prewarm
+
+Created on 2025-03-17
+Implemented on _TBD_
+Author: Alexey Kondratov (@ololobus)
+
+## Summary
+
+This RFC describes an approach to reduce performance degradation due to missing caches after compute node restart, i.e.:
+
+1. Rolling restart of the running instance via 'warm' replica.
+2. Auto-prewarm compute caches after unplanned restart or scale-to-zero.
+
+## Motivation
+
+Neon currently implements several features that guarantee high uptime of compute nodes:
+
+1. Storage high-availability (HA), i.e. each tenant shard has a secondary pageserver location, so we can quickly switch over compute to it in case of primary pageserver failure.
+2. Fast compute provisioning, i.e. we have a fleet of pre-created empty computes, that are ready to serve workload, so restarting unresponsive compute is very fast.
```

Comment:
I want to highlight that we do not automatically restart unresponsive computes and in the case of a pod failure, NeonVM controller restarts the pod which is not going to be fast. It's only fast when using cplane restart compute API.

Reply:
There is a force-suspend computes in cplane after 5 mins of no response to /status. That's what I meant here. NeonVM controller logic I mentioned in 3.